src/clj_kondo/core.clj

```diff
@@ -23,10 +23,12 @@
   subject to change."
   [{:keys [:config :findings :summary :analysis :report-level]}]
   (let [output-cfg (:output config)
-        report-level (some-> report-level keyword)
-        report-level? (or (set (drop-while #(not= report-level %)
-                                           [:info :warning :error]))
-                          (constantly true))
+        report-level? (let [report-level (some-> report-level keyword)]
```

Comment:
can you do an earlier check on report-level so we don't have to do all this work when report-level is nil

Reply:
Yes, but it adds some duplication. Let me try that.